mosviz/controls/slits/rectangle_slit.py

```diff
@@ -0,0 +1,162 @@
+import numpy as np
+
+from regions import RectangleSkyRegion, RectanglePixelRegion, PixCoord
+from astropy.coordinates import Angle, SkyCoord
+from astropy import units as u
+
+from .slit_base import _MOSVizSlit
+
+
+class _RectangleSlitBase(_MOSVizSlit):
```

Comment:
Does this really need to be a `Rectangle`?  I might be missing something but it looks general enough that it could be `_RegionsSlitBase` or similar, couldn't it?  Since the drawing is now fully deferred to `regions`?

Reply:
So for this part it has to be specific for rectangles. This is beacuse some of the functions are specific for rectangle patches (like `self.patch.get_height()` vs `self.patch.get_radius()`).

Reply:
Oh, gotcha.  That makes sense... I think some of the things implemented in this class could go up to `_MOSVizSlit`, then (e.g. `patch` and `has_patch`), since they are generic to any region type, not just rectangles.

*But* there's no driving need to do that right now since we are currently only supporting rectangles.  In a future PR allowing other regions this change could be made.